### PR TITLE
Interface to Snmp proxy through common methods shared by allTango dev…

### PR DESCRIPTION
**Pull request by aleolivo**
…ices that communicate via snmp

### CODE REVIEW

File: src/SnmpClient.cpp [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+	
+	//	Initialize property data members
+	
+	/*----- PROTECTED REGION END -----*/	//	SnmpClient::get_device_property_before
+
+	mandatoryNotDefined = false;
+
+	//	Read device properties from database.
+	Tango::DbData	dev_prop;
+	dev_prop.push_back(Tango::DbDatum("SnmpProxyName"));
```
Comment (abogani) [inline src/SnmpClient.cpp]: Please use "DeviceName" instead of "SnmpProxyName"

File: src/SnmpClient.cpp [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+	get_device_property();
+	
+	//	No longer if mandatory property not set. 
+	if (mandatoryNotDefined)
+		return;
+
+	/*----- PROTECTED REGION ID(SnmpClient::init_device) ENABLED START -----*/
+	
+	//	Initialize device
+	device_proxy = new Tango::DeviceProxy(snmpProxyName);
```
Comment (abogani) [inline src/SnmpClient.cpp]: Put this into a try catch statement.

File: src/SnmpClient.h [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+/*----- PROTECTED REGION END -----*/	//	SnmpClient::Additional Class Declarations
+
+class SnmpClient : public TANGO_BASE_CLASS
+{
+
+/*----- PROTECTED REGION ID(SnmpClient::Data Members) ENABLED START -----*/
+
+//	Add your own data members
+Tango::DeviceProxy *device_proxy;
```
Comment (abogani) [inline src/SnmpClient.h]: indent this

File: src/SnmpClient.h [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+template<typename F, typename T> void convert(const F &reply, T &reply_converted)
+{
+	stringstream conv;
+	conv << reply << flush;
+	if (conv.fail())
+		Tango::Except::throw_exception(
+				"API_ConverionError",
+				"Conversion fail " + conv.str(),
+				"SnmpClient::convert()");
+	conv >> reply_converted;
```
Comment (abogani) [inline src/SnmpClient.h]: This doesn't work if reply_converted is a string with one or more space. You have to define a partial customization of the template "template\<typename F\> void convert(const F &reply, string &reply_converted)"

**Comment by aleolivo** [comment]
Changes requested done!

**Pull request merged**